src/OrchardCore.Modules/OrchardCore.MiniProfiler/Startup.cs

```diff
@@ -33,7 +33,16 @@ public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder ro
             var store = serviceProvider.GetService<IStore>();
 
             var factory = store.Configuration.ConnectionFactory;
-            store.Configuration.ConnectionFactory = new MiniProfilerConnectionFactory(factory);
+            var profilerFactory = new MiniProfilerConnectionFactory(factory);
+            store.Configuration.ConnectionFactory = profilerFactory;
+            var dialect = SqlDialectFactory.For(factory.DbConnectionType);
+            SqlDialectFactory.Register(profilerFactory.DbConnectionType, dialect);
```

Comment:
Can there be an issue if we have tenants that have different databases and have the mini profiler enabled?  Since each tenant will want to register a different dialect?

Reply:
Before there is

            var factory = store.Configuration.ConnectionFactory;
            var profilerFactory = new MiniProfilerConnectionFactory(factory);

So it seems that it only depends on the store, **so seems to be ok** as the store is a **tenant** singleton

Reply:
I am just worried that anywhere inside yessql the dialect is resolved using the connection type, and in this case it's ProfileDbConnection, for all the tenants, even if this type wraps the actual connection type.

Reply:
Yes, it seems that a tenant with sqlite and a tenant running sqlserver each reset the `SqlDialectFactory` `ProfileDbConnection` type as they startup with their own dialects. 

That being said, it all seems to work ok still. And was also happening previously, but based on the magic string `profiledbconnection`. 

So perhaps not getting resolved anywhere inside YesSql?

Reply:
I checked the code and it might be resolved before we change the factory. So that's fine. Also very limited scope since we use it for devs. I might need to change yessql to at least be able to support that just in case. Too much "static" here. So GTG.

I also did all the pending changes.